src/main/java/io/github/opencubicchunks/cubicchunks/core/asm/mixin/fixes/common/vanillaclient/MixinNetHandlerPlayServer.java

```diff
@@ -216,4 +265,22 @@ public void preprocessTeleportConfirm(CPacketConfirmTeleport packetIn, CallbackI
         // TODO: make this faster
         return VanillaNetworkHandler.copyPacket(packetIn);
     }
+
+    @Inject(method = "Lnet/minecraft/network/NetHandlerPlayServer;setPlayerLocation(DDDFFLjava/util/Set;)V",
+            at = @At(value = "INVOKE", shift = At.Shift.BEFORE,
+                    target = "Lnet/minecraft/network/NetHandlerPlayServer;sendPacket(Lnet/minecraft/network/Packet;)V"))
+    private void notifyVanillaHandlerOfTeleport(double x, double y, double z, float yaw, float pitch, Set<SPacketPlayerPosLook.EnumFlags> relativeSet, CallbackInfo ci) {
+        if (!CubicChunksConfig.allowVanillaClients) {
+            return;
+        }
+        World world = this.player.world;
+        if (!((ICubicWorld) world).isCubicWorld()) {
+            return;
+        }
+        VanillaNetworkHandler vanillaHandler = ((ICubicWorldInternal.Server) world).getVanillaNetworkHandler();
+        if (!vanillaHandler.hasCubicChunks(this.player)) {
+            vanillaHandler.updatePlayerPosition((PlayerCubeMap) ((WorldServer) world).getPlayerChunkMap(), this.player,
+                    new CubePos(Coords.blockToCube(this.targetPos.x), Coords.blockToCube(this.targetPos.y), Coords.blockToCube(this.targetPos.z)), this.teleportId);
+        }
```

Comment:
Can you explain why is that in any way necessary?

Reply:
I had a good reason for it (relating to `/tp` not causing chunks to be resent), but I don't recall what the exact circumstances were. I'll double-check what the cause was and get back to you on that.

Reply:
Upon further investigation: it actually wasn't necessary, the actual issue was something else, which I found out after attempting to debug this after removing the `Map<ChunkPos, ChunkPos>`.